### PR TITLE
Prep for 0.2.0 Release

### PR DESCRIPTION
**Pull request by michaeltlombardi**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,17 +1,44 @@
...
+
+- `compatibility` feature and flag (usable only with the `taskscheduler_api2` provider), allowing users to specify which version of Scheduled Tasks the task should be compatible with ([MODULES-6526](https://tickets.puppetlabs.com/browse/MODULES-6526))
+- Documentation of both the legacy `win32_taskscheduler` provider and the new `taskscheduler_api2` into the project readme, ensuring users will not need to refer to Puppet core documentation ([MODULES-6417](https://tickets.puppetlabs.com/browse/MODULES-6417))
+- New helper for `taskscheduler_api2` allowing it to manage scheduled tasks of any compatibility level ([MODULES-6844](https://tickets.puppetlabs.com/browse/MODULES-6844), [MODULES-6845](https://tickets.puppetlabs.com/browse/MODULES-6845))
+
+### Changed
 
-### Summary
+- Default provider from `win32_taskscheduler` to `taskscheduler_api2` ([MODULES-6591](https://tickets.puppetlabs.com/browse/MODULES-6591))
+- Logic for managing triggers, refactoring for improved maintainability ([MODULES-6843](https://tickets.puppetlabs.com/browse/MODULES-6843), [MODULES-6895](https://tickets.puppetlabs.com/browse/MODULES-6895))
```
Comment (michaeltlombardi) [inline CHANGELOG.md]: @Iristyle, @clairecadman, @glennsarti: Not sure if we should keep these ticketed entries describing changes to internals in the changelog. They're significant but mostly internal-facing.

Comment (michaeltlombardi) [reply CHANGELOG.md]: Specifically unsure of lines 12, 17, and 18.

**Comment by michaeltlombardi** [comment]
Still to come: README updates, final update to changelog and metadata prior to release.

**Comment by michaeltlombardi** [comment]
@glennsarti good catch, I failed to update the title to indicate WIP

File: CHANGELOG.md [hunk 2]
```diff
@@ -40,5 +42,6 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/) a
 - Updated README with examples for the new provider ([MODULES-6264](https://tickets.puppetlabs.com/browse/MODULES-6264))
 - Updated acceptance tests for the new provider ([MODULES-6362](https://tickets.puppetlabs.com/browse/MODULES-6362))
 
-[Unreleased]: https://github.com/puppetlabs/puppetlabs-scheduled_task/compare/0.1.0...HEAD
+[Unreleased]: https://github.com/puppetlabs/puppetlabs-scheduled_task/compare/0.2.0...HEAD
```
Comment (michaeltlombardi) [inline CHANGELOG.md]: Note that neither this line nor the next will produce a working link until _after_ we tag for the `0.2.0` release.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,17 +1,47 @@
...
-### Summary
+## [Unreleased]
 
-Initial unsupported release of the scheduled_task module.  Adapts the Puppet scheduled_task resource to use the modern Version 2 API.
+## [0.2.0] - 2018-05-09
+
+### Added
+
+- `compatibility` feature and flag (usable only with the `taskscheduler_api2` provider), allowing users to specify which version of Scheduled Tasks the task should be compatible with ([MODULES-6526](https://tickets.puppetlabs.com/browse/MODULES-6526))
+- Documentation of both the legacy `win32_taskscheduler` provider and the new `taskscheduler_api2` into the project readme, ensuring users will not need to refer to Puppet core documentation ([MODULES-6417](https://tickets.puppetlabs.com/browse/MODULES-6417))
```
Comment (clairecadman) [inline CHANGELOG.md]: Omit "both" and "new". And README?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,17 +1,47 @@
...
+### Changed
+
+- Default provider from `win32_taskscheduler` to `taskscheduler_api2` ([MODULES-6591](https://tickets.puppetlabs.com/browse/MODULES-6591))
+- Logic for managing triggers, refactoring for improved maintainability ([MODULES-6843](https://tickets.puppetlabs.com/browse/MODULES-6843), [MODULES-6895](https://tickets.puppetlabs.com/browse/MODULES-6895))
+- `win32_taskscheduler` to use the adapter code developed for `taskscheduler_api2` for improved maintainability ([MODULES-6845](https://tickets.puppetlabs.com/browse/MODULES-6845))
+
+### Fixed
+
+- Metadata to ensure a correct link on the Puppet Forge, by [@TraGicCode](https://github.com/TraGicCode) in [PR 12](https://github.com/puppetlabs/puppetlabs-scheduled_task/pull/12)
+- Metadata to support only Puppet version `4.9.0` and above as earlier versions do not support translation, which this module uses (MAINT)
```
Comment (clairecadman) [inline CHANGELOG.md]: Comma after above.

File: README.md [hunk 5]
```diff
@@ -206,11 +214,12 @@ Available providers are:
 
 ###### win32_taskscheduler
 
-This provider manages scheduled tasks on Windows using the legacy API.
+This legacy provider manages scheduled tasks on Windows using the v2 api but only manages scheduled tasks whose compatibility level is set to 1 (Windows XP or Windows Server 2003).
+It is a backward compatible update to the provider of the same name in Puppet core and replaces it.
```
[the remark this answers is not in the document]
Comment (michaeltlombardi) [reply README.md]: Updated!

File: CHANGELOG.md (hunk 1 above)
Comment (Iristyle) [reply CHANGELOG.md]: Yeah, I'm fine with it... It's highly unlikely anyone is relying on this in any way, but I'm OK with it being in here for the sake of anyone working on the project.

File: CHANGELOG.md [hunk 6]
```diff
@@ -5,12 +5,35 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/) a
 
 ## [Unreleased]
 
+### Added
+
+- `compatibility` feature and flag (usable only with the `taskscheduler_api2` provider), allowing users to specify which version of Scheduled Tasks the task should be compatible with ([MODULES-6526](https://tickets.puppetlabs.com/browse/MODULES-6526))
```
Comment (Iristyle) [inline CHANGELOG.md]: Can we add "defaults to 1 for backward compatibility" here?

File: README.md [hunk 7]
```diff
@@ -194,8 +194,16 @@ This parameter will not be used to determine if a scheduled task is in sync or n
...
+
 The compatibility level associated with the task.
-May currently only be set to 1 for compatibility with tasks on a Windows XP or Windows Server 2003 computer.
+Can be set to:
+
+- `1` for compatibility with tasks on a Windows XP or Windows Server 2003 computer
+- `2` for compatibility with tasks on a Windows 2008 computer,
+- `3` for compatibility with new features for tasks introduced in Windows 7 and 2008R2
+- `4` for compatibility with new features for tasks introduced in Windows 8, Server 2012R2 and Server 2016
+- `6` for compatibility with new features for tasks introduced in Windows 10
```
Comment (Iristyle) [inline README.md]: I think we might want to add a note about "6" being undocumented and that it's YMMV / not recommended.

I'm not really sure if people should be using it or not, it doesn't show up in the WMI MOFs from what I can tell (at least it's not exposed through `wbemtest`), PowerShell can't get at it properly, etc.

We might also want to link to the docs at MSDN that talk about task scheduler compat stuff https://msdn.microsoft.com/en-us/library/windows/desktop/aa384138(v=vs.85).aspx

Comment (michaeltlombardi) [reply README.md]: I think this makes good sense and is the right place to include such a note, adding now.

Comment (michaeltlombardi) [reply README.md]: Notes added below.

File: README.md [hunk 8]
```diff
@@ -194,8 +194,20 @@ This parameter will not be used to determine if a scheduled task is in sync or n
...
+Can be set to:
+
+- `1` for compatibility with tasks on a Windows XP or Windows Server 2003 computer
+- `2` for compatibility with tasks on a Windows 2008 computer
+- `3` for compatibility with new features for tasks introduced in Windows 7 and 2008R2
+- `4` for compatibility with new features for tasks introduced in Windows 8, Server 2012R2 and Server 2016
+- `6` for compatibility with new features for tasks introduced in Windows 10
+  - **NOTE:** This compatibility setting is _not_ documented and we recommend that you do not use it.
+
+See the [Microsoft documentation on compatibility levels and their differences](https://msdn.microsoft.com/en-us/library/windows/desktop/aa384138(v=vs.85).aspx) for more information.
```
Comment (Iristyle) [inline README.md]: Need to `\(` the `(` and `\)` the `)` in the link

**Pull request merged**